### PR TITLE
[reboot]: Allow reboot to happen regardless vendor hook errors

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

The fix is aimed to resolve reboot issue when vendor reboot hook is executed during `syncd` service startup.

Relevant PRs:
1. https://github.com/sonic-net/sonic-utilities/pull/3203
2. https://github.com/sonic-net/sonic-utilities/pull/819

#### What I did
* Fixed reboot failure in case of vendor reboot hook error

#### How I did it
* Removed error validation

#### How to verify it
1. Execute `reboot` when `syncd` is starting and ASIC reset is in progress

#### Previous command output (if the output of a command-line utility has changed)
* N/A

#### New command output (if the output of a command-line utility has changed)
* N/A

### CODE REVIEW

**Review by stepanblyschak — APPROVED** [review]

**Comment by bingwang-ms** [comment]
@prgeor Can you please help review this PR? Thanks

**Comment by prgeor** [comment]
@nazariig What is the connection between syncd service startup and your platform's reboot hook?

**Comment by nazariig** [comment]
@prgeor this bug is vendor specific. The flow can be interrupted if `cold` reboot is called during `syncd` service startup since the last one triggers ASIC reset. During ASIC reset we cannot read the FW state, thus reboot hook returns error

File: scripts/reboot [hunk 1]
```diff
@@ -278,13 +276,6 @@ fi
 if [ -x ${DEVPATH}/${PLATFORM}/${PRE_REBOOT_HOOK} ]; then
     debug "Executing the pre-reboot script"
     ${DEVPATH}/${PLATFORM}/${PRE_REBOOT_HOOK}
-    EXIT_CODE=$?
```
[the remark this answers is not in the document]
Comment (nazariig) [reply scripts/reboot:281]: @prgeor done

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@bingwang-ms can you take this to 202405 as a bug fix/improvement

**Pull request merged**